feature-libs/quote/components/comments/quote-comments.component.spec.ts

```diff
@@ -0,0 +1,502 @@
+          },
+          {
+            provide: EventService,
+            useValue: mockedEventService,
+          },
+          {
+            provide: QuoteUIConfig,
+            useValue: quoteUiConfig,
+          },
+        ],
+      }).compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(QuoteCommentsComponent);
+    htmlElem = fixture.nativeElement;
+    component = fixture.componentInstance;
+
+    fixture.detectChanges();
+    spyOn(component.commentsComponent, 'resetForm');
+  });
+
+  function initTestData() {
+    quote = createEmptyQuote();
+    quote.code = QUOTE_CODE;
+    quoteUiConfig = {
+      quote: { maxCharsForComments: 5000 },
+    };
+  }
+
+  function initMocks() {
+    mockedQuoteFacade = jasmine.createSpyObj('QuoteFacade', [
+      'getQuoteDetails',
+      'addQuoteComment',
+    ]);
+    asSpy(mockedQuoteFacade.getQuoteDetails).and.returnValue(of(quote));
+    asSpy(mockedQuoteFacade.addQuoteComment).and.returnValue(of({}));
+
+    mockedEventService = jasmine.createSpyObj('EventService', ['dispatch']);
+  }
+
+  function asSpy(f: any) {
+    return <jasmine.Spy>f;
+  }
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('Ghost animation', () => {
+    it('should render view for ghost animation', () => {
+      component.quoteDetails$ = NEVER;
+      fixture.detectChanges();
+
+      CommonQuoteTestUtilsService.expectElementPresent(
+        expect,
+        htmlElem,
+        '.cx-ghost-comment'
+      );
+
+      CommonQuoteTestUtilsService.expectElementPresent(
+        expect,
+        htmlElem,
+        '.cx-ghost-input-title'
+      );
+
+      CommonQuoteTestUtilsService.expectElementPresent(
+        expect,
+        htmlElem,
+        '.cx-ghost-input'
+      );
+    });
+  });
+
+  it('should render the messaging section by default', () => {
+    CommonQuoteTestUtilsService.expectElementPresent(
+      expect,
+      htmlElem,
+      'cx-messaging'
+    );
+  });
+
+  it('should collapse the comments area when clicking the toggle', () => {
+    CommonQuoteTestUtilsService.clickToggle(htmlElem, false);
+    fixture.detectChanges();
+    CommonQuoteTestUtilsService.expectElementNotPresent(
+      expect,
+      htmlElem,
+      'cx-messaging'
+    );
+  });
+
+  it('should toggle the comments on enter', () => {
+    CommonQuoteTestUtilsService.clickToggle(htmlElem, true);
+    fixture.detectChanges();
+    CommonQuoteTestUtilsService.expectElementNotPresent(
+      expect,
+      htmlElem,
+      'cx-messaging'
+    );
+  });
+
+  it('should expand the comments area when clicking the toggle', () => {
+    component.expandComments = false;
+    CommonQuoteTestUtilsService.clickToggle(htmlElem, false);
+    CommonQuoteTestUtilsService.expectElementPresent(
+      expect,
+      htmlElem,
+      'cx-messaging'
+    );
+  });
+
+  it('should pipe empty quote comments to empty message events', () => {
```

Comment:
Please, add `describe('pipe', () => {...})` around tests for piping quote messages

Reply:
@Uli-Tiger, could you add a describe?